tests/unit/test_utils.py

```diff
@@ -146,9 +141,96 @@ def test_cdn_handle_domain_updates_with_changes(
 
         dns.add_cname("_acme-challenge.moo.com")
         domain_names = ["moo.com"]
-        (updated_domain_names, no_domain_updates) = handle_domain_updates(
+
+        updated_domain_names = handle_domain_updates(
             dict(domains=domain_names), instance
         )
-
-        assert no_domain_updates == False
         assert updated_domain_names == domain_names
+
+
+@pytest.mark.parametrize(
+    "factory",
+    [
+        factories.ALBServiceInstanceFactory,
+        factories.CDNServiceInstanceFactory,
+        factories.CDNDedicatedWAFServiceInstanceFactory,
+        factories.DedicatedALBServiceInstanceFactory,
+    ],
+)
+def test_handle_domain_errors_on_missing_dns(
+    no_context_clean_db, no_context_app, dns, factory
+):
+    with no_context_app.app_context():
+        dns.add_cname("_acme-challenge.moo.com")
+        domain_names = ["moo.com", "cow.com"]
```

Comment:
We should use only official example domains or domains we control for test data - can we use something like `moo.example.com` and `cow.example.net`